src/plugins/walletConnectToDapps/components/modals/SessionProposalOverview.tsx

```diff
@@ -0,0 +1,299 @@
+  fontSize: 'sm' as const,
+  color: 'cyan.600',
+  fontWeight: 'semibold' as const,
+}
+
+const infoIconStyles = {
+  boxSize: 4,
+  color: 'cyan.600',
+  strokeWidth: 2,
+  ml: 'auto',
+}
+
+const containerBoxStyles = {
+  bg: 'transparent',
+  borderTopRadius: '24px',
+  borderTop: '1px solid',
+  borderLeft: '1px solid',
+  borderRight: '1px solid',
+  px: 8,
+  py: 4,
+  mx: -6,
+  mb: -6,
+  mt: 4,
+}
+
+const mainHStackStyles = {
+  spacing: 4,
+  w: 'full' as const,
+  justify: 'space-between' as const,
+  align: 'start' as const,
+}
+
+const leftHStackStyles = {
+  spacing: 3,
+  align: 'start' as const,
+  flex: 1,
+}
+
+const avatarStyles = {
+  boxSize: '32px',
+  borderRadius: 'full',
+}
+
+const textVStackStyles = {
+  spacing: 1,
+  align: 'start' as const,
+  h: '32px',
+  justify: 'space-between' as const,
+}
+
+const labelTextStyles = {
+  fontSize: 'xs' as const,
+  color: 'text.subtle',
+  fontWeight: 'medium' as const,
+  lineHeight: '1',
+}
+
+const selectorHStackStyles = {
+  spacing: 3,
+  align: 'center' as const,
+  h: '20px',
+}
+
+const networkVStackStyles = {
+  spacing: 1,
+  align: 'end' as const,
+  h: '32px',
+  justify: 'space-between' as const,
+}
+
+const networkHStackStyles = {
+  spacing: 2,
+  align: 'center' as const,
+  h: '20px',
+}
+
+const addressTextStyles = {
+  fontSize: 'sm' as const,
+  fontWeight: 'medium' as const,
+}
+
+const arrowIconStyles = {
+  color: 'text.subtle',
+  boxSize: 3,
+}
+
+const chainIconStyles = {
+  boxSize: 5,
+}
+
+const chainCounterStyles = {
+  size: 5,
+  bg: 'gray.100',
+  color: 'text.base',
+  fontSize: '2xs' as const,
+  fontWeight: 'medium' as const,
+  ml: -1.5,
+}
+
+const buttonsHStackStyles = {
+  spacing: 4,
+  w: 'full' as const,
+  mt: 4,
+}
+
+const buttonStyles = {
+  size: 'lg' as const,
+  flex: 1,
+}
+
+type SessionProposalOverviewProps = {
+  modalBody: JSX.Element
+  selectedAccountNumber: number | null
+  uniqueAccountNumbers: number[]
+  selectedNetworks: ChainId[]
+  onAccountClick: () => void
+  onNetworkClick: () => void
+  onConnectSelected: () => void
+  onReject: () => void
+  isLoading: boolean
+  canConnect: boolean
+  translate: (key: string) => string
+}
+
+export const SessionProposalOverview: React.FC<SessionProposalOverviewProps> = ({
+  modalBody,
+  selectedAccountNumber,
+  uniqueAccountNumbers,
+  selectedNetworks,
+  onAccountClick,
+  onNetworkClick,
+  onConnectSelected,
+  onReject,
+  isLoading,
+  canConnect,
+  translate,
+}) => {
+  const assetsById = useAppSelector(selectAssets)
+  const accountIdsByAccountNumberAndChainId = useAppSelector(
+    selectAccountIdsByAccountNumberAndChainId,
+  )
+  const chainAdapterManager = getChainAdapterManager()
+  const borderColor = useColorModeValue('gray.100', 'whiteAlpha.100')
+
+  const hasMultipleAccounts = uniqueAccountNumbers.length > 1
+  const visibleChains = selectedNetworks.slice(0, maxVisibleChains)
+  const remainingCount = selectedNetworks.length - maxVisibleChains
```

Comment:
logic already exists in receive - probably worth consolidating visual bits / smol logic into a smol reused component

Reply:
on second though, meh